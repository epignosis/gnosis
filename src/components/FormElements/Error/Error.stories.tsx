import React from "react";
import { Story } from "@storybook/react";
import ErrorComponent from "./Error";
import { Input } from "@components";

export default {
  title: "components/Form Elements/Error",
  component: ErrorComponent,
  parameters: {
    controls: { hideNoControlsWarning: true },
  },
};

export const Error: Story = () => (
  <ErrorComponent>
    <p>
      You entered an incorrect email,
      <br /> password, or both!
      <br />
      <br />
      Try again or contact your administrator
    </p>
  </ErrorComponent>
);

export const InputError: Story = () => (
  <div style={{ maxWidth: 500 }}>
    <Input
      id="inline-error-example"
      label="Attention"
      status="error"
      placeholder="Pay attention to the error below"
    />
    <ErrorComponent.InputError>Some error</ErrorComponent.InputError>
  </div>
);
