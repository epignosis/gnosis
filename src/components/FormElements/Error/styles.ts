import { css, Theme, SerializedStyles } from "@emotion/react";
import { hexToRGBA } from "@theme/default/colors";

export const errorContainer = ({ typeScaleSizes, formElements }: Theme): SerializedStyles => css`
  font-size: ${typeScaleSizes.md};
  width: 100%;
  display: inline-block;
  background-color: ${hexToRGBA(formElements.error.background, 0.25)};
  border: 1px solid ${formElements.error.borderColor};
  border-radius: 2px;
  padding: 1.25rem 2rem 1.5rem;
`;

export const inputError = ({ formElements }: Theme): SerializedStyles => css`
  color: ${formElements.input.errorColor};
  margin-left: 0.25rem;
`;
