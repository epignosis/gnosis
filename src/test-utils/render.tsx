import React, { FC, ReactElement, ComponentType } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options });

export * from "@testing-library/react";

export { customRender as render };
