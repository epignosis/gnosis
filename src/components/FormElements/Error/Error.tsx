import React, { FC } from "react";
import { errorContainer } from "./styles";

const Error: FC = ({ children }) => {
  return <article css={errorContainer}>{children}</article>;
};

export default Error;
