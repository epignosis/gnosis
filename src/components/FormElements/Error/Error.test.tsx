import React from "react";
import faker from "faker";
import Error from "./Error";
import { render, screen } from "@test-utils/render";

describe("Forms: <Error />", () => {
  it("render correctly", () => {
    const txt = faker.lorem.sentence();
    render(<Error>{txt}</Error>);
    const errorContainer = screen.getByText(txt);

    expect(errorContainer).toHaveTextContent(txt);
  });

  it("matches snapshot", () => {
    const { container } = render(<Error>Something went wrong</Error>);

    expect(container).toMatchSnapshot();
  });
});

describe("Forms: <Error.InputError />", () => {
  it("renders correctly", () => {
    const txt = faker.lorem.sentence();
    render(<Error.InputError>{txt}</Error.InputError>);
    const errorSpan = screen.getByText(txt);

    expect(errorSpan).toHaveTextContent(txt);
  });

  it("matches snapshot", () => {
    const { container } = render(<Error.InputError>Something went wrong</Error.InputError>);

    expect(container).toMatchSnapshot();
  });
});
