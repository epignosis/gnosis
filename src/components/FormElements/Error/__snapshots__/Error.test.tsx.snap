// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forms: <Error /> matches snapshot 1`] = `
<div>
  <article
    class="css-1g7t4ds-errorContainer-errorContainer"
  >
    Something went wrong
  </article>
</div>
`;

exports[`Forms: <Error.InputError /> matches snapshot 1`] = `
<div>
  <span
    class="css-1wc9q6w-inputError-inputError"
  >
    Something went wrong
  </span>
</div>
`;
